cql/src/main/java/org/apache/cassandra/stargate/transport/internal/RequestToHeadersMapper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.stargate.transport.internal;
+
+import java.net.InetSocketAddress;
+import java.util.*;
+
+public class RequestToHeadersMapper {
+  public static final String TENANT_ID_HEADER_NAME = "tenant_id";
+
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  public static Map<String, String> toHeaders(
+      Optional<InetSocketAddress> sourceAddressWithEncodedTenantId) {
```

Comment:
From my POV it would be more intuitive to use `ClientInfo` as a parameter (or the extracted ID `String`) since the way we determine tenant ID is very specific to how `ClientInfo` is constructed. Any `InetSocketAddress` would be be suitable.

Reply:
done